File: Python/student_management_system/data_management.py
import json
from student_class import Student


def save_students(filename):
    unique_student_ids = set()
    unique_students = []

    for student in Student.all_students:
        student_id = student.generate_student_ID()
        if student_id not in unique_student_ids:
            unique_student_ids.add(student_id)
            unique_students.append(student)

    with open(filename, "w") as file:
        json.dump([student.to_dict()
                  for student in unique_students], file, indent=4)


def load_students(filename):
    with open(filename, "r") as file:
        student_data_list = json.load(file)

    return [Student.from_dict(student_data) for student_data in student_data_list]

File: JavaScript/student_management_system/dataManagement.js
import { writeFileSync, readFileSync } from "fs";
import Student from "./studentClass.js";

export function saveStudents(filename) {
    const uniqueStudentIds = new Set();
    const uniqueStudents = [];

    Student.allStudents.forEach((student) => {
        const studentId = student.generateStudentID();
        if (!uniqueStudentIds.has(studentId)) {
            uniqueStudentIds.add(studentId);
            uniqueStudents.push(student);
        }
    });

    const studentData = uniqueStudents.map((student) => student.toDict());
    writeFileSync(filename, JSON.stringify(studentData, null, 4), "utf8");
}

export function loadStudents(filename) {
    const studentDataList = JSON.parse(readFileSync(filename, "utf8"));

    return studentDataList.map((studentData) => Student.fromDict(studentData));
}
